conversion from 'long' to 'unsigned int' requires a narrowing conversion on MSVC

This was previously a warning but now HPX actually fails to compile because of this:
```console
C:\Repos\hpx\libs\format\src\format.cpp(83,17): error C2397:  conversion from 'long' to 'unsigned int' requires a narrowing conversion [C:\Repos\hpx\cmake-build-debug\libs\format\hpx_format.vcxproj]
C:\Repos\hpx\libs\format\src\format.cpp(88,21): error C2397:  conversion from 'long' to 'unsigned int' requires a narrowing conversion [C:\Repos\hpx\cmake-build-debug\libs\format\hpx_format.vcxproj]
C:\Repos\hpx\libs\format\src\format.cpp(96,21): error C2397:  conversion from 'long' to 'unsigned int' requires a narrowing conversion [C:\Repos\hpx\cmake-build-debug\libs\format\hpx_format.vcxproj]
...
```
* Configuration:
    * Commit: 4ab69844795a2674dff977509fe09588761b8e6e
    * Windows 10.0.17763.615
    * Visual Studio 2019 x64 (MSVC 19.21.27702.2)
    * CMake 3.14.19050301-MSVC_2
    * Boost 1.70.0
    * Microsoft/vcpkg@f9fedc930cf36ce6cec69a3d09879b3c7f23eb0b
* Full log:
    * [diag.log](https://github.com/STEllAR-GROUP/hpx/files/3407452/diag.log)

--- a/libs/assertion/include/hpx/assertion.hpp
+++ b/libs/assertion/include/hpx/assertion.hpp
@@ -54,8 +54,8 @@ namespace hpx { namespace assertion {
 #define HPX_ASSERT_(expr, msg)                                                 \
     ::hpx::assertion::detail::evaluate_assert(                                 \
         [&]() -> bool { return !!(expr); },                                    \
-        ::hpx::assertion::source_location{                                     \
-            __FILE__, __LINE__, HPX_ASSERT_CURRENT_FUNCTION},                  \
+        ::hpx::assertion::source_location{__FILE__,                            \
+            static_cast<unsigned>(__LINE__), HPX_ASSERT_CURRENT_FUNCTION},     \
         HPX_PP_STRINGIZE(expr), [&]() { return msg; });                        \
 /**/
 

